File: hackhive-frontend/src/services/studentWorkspacePreferenceService.js
import api from "../api/axios";

export const studentWorkspacePreferenceService = {
    getPreferences: async () => {
        const response = await api.get("/student/workspace-preferences");
        return response.data?.data ?? response.data;
    },

    updatePreferences: async (payload) => {
        const response = await api.put("/student/workspace-preferences", payload);
        return response.data?.data ?? response.data;
    },
};

export default studentWorkspacePreferenceService;

File: hackhive-backend/src/main/java/com/hackhive/student/service/StudentWorkspacePreferenceService.java
package com.hackhive.student.service;

import com.hackhive.student.dto.request.UpdateStudentWorkspacePreferenceRequest;
import com.hackhive.student.dto.response.StudentWorkspacePreferenceResponse;
import com.hackhive.student.entity.StudentProfile;
import com.hackhive.student.entity.StudentWorkspacePreference;

public interface StudentWorkspacePreferenceService {

    StudentWorkspacePreferenceResponse getPreferences();

    StudentWorkspacePreferenceResponse updatePreferences(UpdateStudentWorkspacePreferenceRequest request);

    StudentWorkspacePreference createDefaultPreferences(StudentProfile studentProfile);
}
